lib/srv/regular/sshserver.go

```diff
@@ -242,6 +245,10 @@ type Server struct {
 	proxySigner PROXYHeaderSigner
 	// caGetter is used to get host CA of the cluster to verify signed PROXY headers
 	caGetter CertAuthorityGetter
+
+	// remoteForwardingMap holds the remote port forwarding listeners that need
+	// to be closed when forwarding finishes, keyed by listen addr.
+	remoteForwardingMap utils.SyncMap[string, io.Closer]
```

Comment:
Is there some extra synchronization that prevents new additions to `remoteForwardingMap` while the `*Server` is being closed or is closed?